apps/email/js/cards/compose.js

```diff
@@ -551,13 +599,13 @@ return [
           makeBubble = true;
           break;
       }
+
       if (makeBubble) {
-        // TODO: Need to match the email with contact name.
-        node.style.width = '0.5rem';
-        var mailbox = model.api.parseMailbox(node.value);
-        this.insertBubble(node, mailbox.name, mailbox.address);
-        node.value = '';
+        entryMatch = this.autocomplete.topMatch ||
```

Comment:
The automatic use of topMatch here works out, I think, but the emergent control flow is a little weird, and I think it'd be good to document the intentional behaviour in a comment or markdown file or on the wiki somewhere.  (At your discretion.  Mainly there are situations where someone will file a bug and at least my exact recall of what led to that situation has faded somewhat and I don't want to do the bugzilla legwork to find out.  I had started https://wiki.mozilla.org/Gaia/Email/UX/Decisions at one point, that was mainly for cases like the sub-folder icon asymmetry that I thought was a weird choice and wanted to be able to point people.  Or maybe it's not worth the effort.  Your call!)

Specifically, we're causing command and semicolon act like "tab" might in a lot of autocomplete boxes, with it triggering the selection and clearing the input.  This works great.  But the control flow is also considering doing this when the user types the space character.  We don't make a bubble because the user hasn't typed an "@".  I think this will continue to be fine, but it's awkward, so as alluded to above, I think some type of comment acknowledging this might be in order if we don't want to clean up the logic.  (Not sure it's worth cleaning up the logic.)